owrx/websocket.py

```diff
@@ -175,60 +177,67 @@ def read_loop(self):
         def protected_read(num):
             data = self.handler.rfile.read(num)
             if data is None:
-                raise Drained()
-            if len(data) != num:
+                data = bytes()
+            while self.open and len(data) < num:
+                    (read, _, _) = select.select([self.interruptPipeRecv, self.handler.rfile], [], [], 15)
+                    if self.handler.rfile in read:
+                        data += self.handler.rfile.read(num - len(data))
+                    else:
+                        if len(data) == 0:
+                            raise InterruptOrTimeout()
+                        else:
+                            raise IncompleteRead()
+            if len(data) < num:
                 raise IncompleteRead()
             return data
 
         self.open = True
         while self.open:
-            (read, _, _) = select.select([self.interruptPipeRecv, self.handler.rfile], [], [], 15)
-            if self.handler.rfile in read:
-                available = True
+            try:
+                header = protected_read(2)
                 self.resetPing()
```

Comment:
i'd suggest moving this to `protected_read`, too - it should be reset on every bit of data that comes from the client.